### DISCUSSION TITLE
Cake 4.0.0 installation does not seems to work and show i get run dotnet tool restore message event if it's installed

### DISCUSSION THREAD

**Discussion by jobou363** [cake-build/cake#4311] [Q&A] [answered]
I have cake 3.2.0 and it works but I try to upgrade cake to 4.0.0 and for a reason i don't understand the dotnet dotnet-cake does not work.

I run the commande to update cake using .net 8 (8.0.204) commande line like this :

`dotnet tool install Cake.Tool --version 4.0.0`

I receive a message saying the commande is installed : 
_You can invoke the tool from this directory using the following commands: 'dotnet tool run dotnet-cake' or 'dotnet dotnet-cake'.
Tool 'cake.tool' (version '4.0.0') was successfully installed. Entry is added to the manifest file D:\Git\SAI\SAIADNET\CALCULS\sai.alc\.config\dotnet-tools.json._ 

but when i run the usual commande to run cake

`dotnet tool run dotnet-cake`
or
`dotnet dotnet-cake`

I get the message like it's not installed :
`Run "dotnet tool restore" to make the "dotnet-cake" command available.`

I cannot find any clues anywhere after looking in issues, discussion, stackoverflow.

**Comment by jobou363** [comment, marked as answer]
The problem is related to my windows DevDrive and toolResolverCache.

I had install the version 3 and 3.2 before adding the dev drive so it's probably the reason why it was working.

Just found the solution by deleting toolResolverCache seems to fix the issue. On linux it's at ~/.dotnet/toolResolverCache

Found this here : [github issue](https://github.com/dotnet/sdk/issues/11432)

**Comment by Williams23678** [comment]
// Install Shodan as a Cake Addin
#addin nuget:?package=Shodan&version=2.0.3

// Install Shodan as a Cake Tool
#tool nuget:?package=Shodan&version=2.0.3
